### DISCUSSION TITLE
Installing warpx on a new cluster, and a few general questions

### DISCUSSION THREAD

**Discussion by jinze-liu** [BLAST-WarpX/warpx#4918] [General]
I've successfully installed WarpX on a new cluster and have been able to run most of the example codes, although some still produce errors. I'd like to share the installation process, but I'm actually not sure how to start sharing.

During this period, I will temporarily calculate some minor problems, such as the properties of various waves in a plasma under an external perturbative electric or magnetic field. Is this an area where WarpX excels? In fact, I have just entered this field, and it will take me a long time to understand how WarpX works!

**Comment by RemiLehe** [comment]
Yes, feel free to post your installation process here. We can try to help you debug this. 

Regarding the properties of various waves in plasma (I assume that by "properties" you mean e.g. the dispersion relation), WarpX can indeed model this. Whether or not it is the most efficient tool depends a lot on the regime that you would like to model. (For instance, WarpX takes into account all kinetic effects, but if you know that you are modeling a regime in which the fluid approximation is valid, there are probably other tools.)

More details on WarpX's algorithm can be found in the theory section: 
https://warpx.readthedocs.io/en/latest/theory/intro.html

Here are some examples on how to simulate specific waves in plasmas:
https://warpx.readthedocs.io/en/latest/usage/examples/langmuir/README.html
https://warpx.readthedocs.io/en/latest/usage/examples/ohm_solver_EM_modes/README.html

**Reply by jinze-liu** [reply]
Oh, you understood correctly, that's what I wanted to focus on the dispersion relation, currently I want to add a simple cosine perturbation to the initial electric field, but the setup using the inputable file was successful, the setup using the python script was unsuccessful, the code is as follows:

```
# Define and add the external electric field
external_Efield = picmi.AnalyticAppliedField(
    Ex_expression="1.0*cos(300*z)",
    Ey_expression="0.0",
    Ez_expression="0.0"
)
sim.add_applied_field(external_Efield)  # Correct method to add applied fields

# Define and add diagnostics
field_diag = picmi.FieldDiagnostic(
    name='field_diag',
    grid=grid,
    period=50,
    data_list=['Ex', 'Ey', 'Ez', 'Bx', 'By', 'Bz']
)
sim.add_diagnostic(field_diag)
```

At the same time I need to diagnose the electromagnetic fields data at a fixed grid point within a 3D grid, tried several approaches but was unsuccessful, is there a corresponding example file? Thanks for the help!
